modules/ml/src/main/java/org/apache/ignite/ml/naivebayes/gaussian/GaussianNaiveBayesTrainer.java

```diff
@@ -0,0 +1,297 @@
+import org.apache.ignite.ml.math.util.MapUtil;
+import org.apache.ignite.ml.structures.LabeledVector;
+import org.apache.ignite.ml.structures.LabeledVectorSet;
+import org.apache.ignite.ml.structures.partition.LabeledDatasetPartitionDataBuilderOnHeap;
+import org.apache.ignite.ml.trainers.SingleLabelDatasetTrainer;
+
+/**
+ * Trainer for the naive Bayes classification model.
+ */
+public class GaussianNaiveBayesTrainer extends SingleLabelDatasetTrainer<GaussianNaiveBayesModel> {
+
+    private double[] priorProbabilities;
+    private boolean equiprobableClasses;
+
+    /**
+     * Trains model based on the specified data.
+     *
+     * @param datasetBuilder Dataset builder.
+     * @param featureExtractor Feature extractor.
+     * @param lbExtractor Label extractor.
+     * @return Model.
+     */
+    @Override public <K, V> GaussianNaiveBayesModel fit(DatasetBuilder<K, V> datasetBuilder,
+        IgniteBiFunction<K, V, Vector> featureExtractor, IgniteBiFunction<K, V, Double> lbExtractor) {
+        return updateModel(null, datasetBuilder, featureExtractor, lbExtractor);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected boolean checkState(GaussianNaiveBayesModel mdl) {
+        return true;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected <K, V> GaussianNaiveBayesModel updateModel(GaussianNaiveBayesModel mdl,
+        DatasetBuilder<K, V> datasetBuilder, IgniteBiFunction<K, V, Vector> featureExtractor,
+        IgniteBiFunction<K, V, Double> lbExtractor) {
+        assert datasetBuilder != null;
+
+        PartitionDataBuilder<K, V, EmptyContext, LabeledVectorSet<Double, LabeledVector>> partDataBuilder
+            = new LabeledDatasetPartitionDataBuilderOnHeap<>(
+            featureExtractor,
+            lbExtractor
+        );
+
+        try (Dataset<EmptyContext, LabeledVectorSet<Double, LabeledVector>> dataset = datasetBuilder.build(
+            (upstream, upstreamSize) -> new EmptyContext(),
+            partDataBuilder
+        )) {
+            MeanHelper mean = computeMeans(dataset);
+
+            List<Double> sortedFeatures = new ArrayList<>(mean.featureCount.keySet());
+            sortedFeatures.sort(Double::compareTo);
+
+            int labelCount = sortedFeatures.size();
+            int featureCount = mean.featureSum.get(sortedFeatures.get(0)).length;
+
+            double[][] means = new double[labelCount][featureCount];
+            double[][] variances = new double[labelCount][featureCount];
+            double[] classProbabilities = new double[labelCount];
+            double[] labels = new double[labelCount];
+
+            long datasetSize = mean.featureCount.values().stream().mapToInt(i -> i).sum();
+            Map<Double, double[]> meansHolder = new HashMap<>();
+
+            int lbl = 0;
+            for (Double label : sortedFeatures) {
+                int count = mean.featureCount.get(label);
+                double[] tmp = mean.featureSum.get(label);
```

Comment:
maybe rename to featureValuesSum?

Reply:
`featureSum` was renamed to `featureSumsPerLbl`.